app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -187,18 +190,33 @@ class SubscriptionController @Inject()
   val showSubscriptionComplete: Action[AnyContent] = AuthorisedWithSubscribingAgentAsync() {
     implicit authContext =>
       implicit request => {
-        Future successful{
           val agencyData = for {
             agencyName <- request.flash.get("agencyName")
             arn <- request.flash.get("arn")
           } yield (agencyName, arn)
 
-          agencyData.map(data =>
-            Ok(html.subscription_complete(appConfig.agentServicesAccountUrl, data._1, data._2))
-          ) getOrElse sessionMissingRedirect()
-        }
+          sessionStoreService.fetchContinueUrl.map {_ match {
```

Comment:
@cavster The continue url will be provided in the continue param of agent subscription root or start endpoints. Therefore, it can accept any validated urls.